scheduler-driver/src/test/java/gov/nasa/jpl/aerie/scheduler/SerializedValueSubsetTest.java

```diff
@@ -0,0 +1,89 @@
+                Map.of("C", SerializedValue.of(14), "D", SerializedValue.of(15))
+            )));
+
+    final var subsetMap = SerializedValue.of(
+        Map.of(
+            "B", SerializedValue.of(
+                Map.of("D", SerializedValue.of(15))
+            )));
+    final var notSubsetMap = SerializedValue.of(
+        Map.of(
+            "B", SerializedValue.of(
+                Map.of("D", SerializedValue.of(7))
+            )));
+    assertTrue(subsetOrEqual(reference, subsetMap));
+    assertFalse(subsetOrEqual(reference, notSubsetMap));
+  }
+
+  @Test
+  public void testSubsetEqualityList(){
+    final var reference = SerializedValue.of(List.of(SerializedValue.of(8.0), SerializedValue.of(6.0), SerializedValue.of(9.0)));
+    final var subsetList = SerializedValue.of(List.of(SerializedValue.of(8.0),SerializedValue.of(6.0)));
+    final var nullList = SerializedValue.of(List.of(SerializedValue.NULL));
+    final var notSubsetList = SerializedValue.of(List.of(SerializedValue.of(8.0),SerializedValue.of(7.0)));
+    assertTrue(subsetOrEqual(reference, subsetList));
+    assertTrue(subsetOrEqual(reference, nullList));
+    assertFalse(subsetOrEqual(reference, notSubsetList));
+  }
+  @Test
+  public void testSubsetEqualityValuesInteger(){
+    final var reference = SerializedValue.of(1111);
+    final var sameValue = SerializedValue.of(1111);
+    final var differentValue = SerializedValue.of(1112);
+    assertTrue(subsetOrEqual(reference, sameValue));
+    assertFalse(subsetOrEqual(reference, differentValue));
+  }
+
+  @Test
+  public void testSubsetEqualityValuesBoolean(){
+    final var reference = SerializedValue.of(true);
+    final var sameValue = SerializedValue.of(true);
+    final var differentValue = SerializedValue.of(false);
+    final var differentType = SerializedValue.of(11);
+    assertTrue(subsetOrEqual(reference, sameValue));
+    assertFalse(subsetOrEqual(reference, differentValue));
+    assertFalse(subsetOrEqual(reference, differentType));
+  }
+
+  @Test
+  public void testSubsetEqualityValuesDecimal(){
+    final var reference = SerializedValue.of(BigDecimal.valueOf(122.56));
+    final var sameValue = SerializedValue.of(BigDecimal.valueOf(122.56));
+    final var differentValue = SerializedValue.of(BigDecimal.valueOf(122.57));
+    final var differentType = SerializedValue.of(11);
+    assertTrue(subsetOrEqual(reference, sameValue));
+    assertFalse(subsetOrEqual(reference, differentValue));
+    assertFalse(subsetOrEqual(reference, differentType));
+  }
+
+  @Test
+  public void testSubsetEqualityValuesString(){
+    final var reference = SerializedValue.of("AbCd");
+    final var sameValue = SerializedValue.of("AbCd");
+    final var differentValue = SerializedValue.of("bCd");
+    final var differentType = SerializedValue.of(11);
+    assertTrue(subsetOrEqual(reference, sameValue));
+    assertFalse(subsetOrEqual(reference, differentValue));
+    assertFalse(subsetOrEqual(reference, differentType));
```

Comment:
Can we add a `differentCase` here since we specifically want that to be `false`?
```suggestion
    final var reference = SerializedValue.of("AbCd");
    final var sameValue = SerializedValue.of("AbCd");
    final var differentValue = SerializedValue.of("bCd");
    final var differentCase = SeriealizeValue.of("abcd");
    final var differentType = SerializedValue.of(11);
    assertTrue(subsetOrEqual(reference, sameValue));
    assertFalse(subsetOrEqual(reference, differentValue));
    assertFalse(subsetOrEqual(reference, differentCase));
    assertFalse(subsetOrEqual(reference, differentType));
```

Reply:
done